src/utils/hooks/useYoutube.js

```diff
@@ -1,52 +1,49 @@
-import { useState, useEffect } from 'react';
+import { useEffect } from 'react';
 import axios from 'axios';
 import videosMock from '../../data/youtube-videos-mock.json';
+import { useMainContext } from '../../state/MainProvider';
 
 function useYoutube(searchTerm = 'wizeline') {
-  const [videos, setVideos] = useState(null);
-  const [search, setSearch] = useState('wizeline');
+  const { state, dispatch } = useMainContext();
 
-  useEffect(() => {
-    async function findVideos() {
-      try {
-        const startingUrlResult = await axios.get(
-          `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&q=${searchTerm}&type=video&key=AIzaSyAuaWJQySwY1UL_VKl2OFCteTTaSdSCoac`
-        );
-        setVideos(startingUrlResult.data.items);
-      } catch (error) {
-        console.error('Error fetching videos: ', error);
-        setVideos(
-          videosMock.items
-            .filter(({ id }) => {
-              return id.kind !== 'youtube#channel';
-            })
-            .slice(0, 12)
-        );
-        setSearch(searchTerm);
-      }
-    }
-    findVideos();
-  }, [searchTerm]);
-
-  const searchNewVideo = async (newSearchTerm) => {
+  async function searchVideos(newSearch) {
     try {
       const startingUrlResult = await axios.get(
-        `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&q=${newSearchTerm}&type=video&key=AIzaSyAuaWJQySwY1UL_VKl2OFCteTTaSdSCoac`
+        `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&q=${newSearch}&type=video&key=${process.env.REACT_APP_YOUTUBE_KEY}`
       );
-      setVideos(startingUrlResult.data.items);
+      dispatch({
+        type: 'CHANGE_VIDEOS',
+        payload: startingUrlResult.data.items,
+      });
+      dispatch({
+        type: 'CHANGE_SEARCH',
+        payload: newSearch,
+      });
     } catch (error) {
-      console.error('Error fetching videos: ', error);
-      setVideos(
-        videosMock.items
+      dispatch({
+        type: 'CHANGE_VIDEOS',
+        payload: videosMock.items
           .filter(({ id }) => {
             return id.kind !== 'youtube#channel';
           })
-          .slice(0, 12)
-      );
-      setSearch(newSearchTerm);
+          .slice(0, 12),
+      });
+      dispatch({
+        type: 'CHANGE_SEARCH',
+        payload: newSearch,
+      });
     }
-  };
-  return { videos, searchNewVideo, search };
+  }
+
+  useEffect(() => {
+    if (state.videos.length === 0) {
+      searchVideos(searchTerm);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm]);
+
+  // state is returned for testing purposes
+  return { searchVideos, state };
```

Comment:
Also avoid returning the state here, you should have always a single source of truth for it, which in your case is your MainProvider.